### PR TITLE
removed reference of TableClient instance from LogReplay 

### PR DESCRIPTION
**Pull request by rachana-uniyal**
Removed reference of TableClient instance from LogReplay and updated the methods using tableClient to have it as an argument issue #2641

#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

### CODE REVIEW

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java [hunk 1]
```diff
@@ -101,22 +101,20 @@ public static StructType getAddRemoveReadSchema(boolean shouldReadStats) {
...
-    private final TableClient tableClient;
     private final Tuple2<Protocol, Metadata> protocolAndMetadata;
 
     public LogReplay(
             Path logPath,
             Path dataPath,
             long snapshotVersion,
-            TableClient tableClient,
             LogSegment logSegment,
+            TableClient tableClient,
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java:111]: Make the `TableClient` as the first argument here and all below functions.

**Review by vkorukanti — CHANGES_REQUESTED** [review]
Thanks for the PR. Needs updating the places that call these methods.

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java (hunk 1 above)
Comment (rachana-uniyal) [reply kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java]: Sure, Thanks for the Info.

**Comment by vkorukanti** [comment]
Fixed as part of #3165

**Pull request closed without merging**